cpp/src/arrow/adapters/orc/adapter.cc

```diff
@@ -80,6 +74,12 @@ namespace liborc = orc;
   }                                            \
   catch (const liborc::NotImplementedYet& e) { \
     return Status::NotImplemented(e.what());   \
+  }                                            \
+  catch (const std::exception& e) {            \
```

Comment:
I've added this. WDYT? @pitrou @jorisvandenbossche

Reply:
Perhaps make this `UnknownError` as well?